packages/amplication-server/src/core/resource/resource.service.ts

```diff
@@ -373,6 +379,192 @@ export class ResourceService {
     return resource;
   }
 
+  async createResourceEntitiesFromExistingResource(
+    args: CreateResourceEntitiesArgs,
+    user: User
+  ): Promise<Resource> {
+    const { targetResourceId, entitiesToCopy } = args.data;
+    const resource = await this.prisma.resource.findUnique({
+      where: {
+        id: targetResourceId,
+      },
+      include: {
+        entities: {
+          where: {
+            deletedAt: null,
+          },
+        },
+      },
+    });
+
+    if (!resource) throw new AmplicationError("Target Resource doesn't exist");
+
+    entitiesToCopy.map((entity) => {
+      if (resource.entities.find((x) => x.name === entity.name)) {
+        throw new AmplicationError(
+          `Entity ${entity.name} is already exist in resourceId: ${resource.id}, process abort`
+        );
+      }
+    });
+
+    // 1.create resource entities
+
```

Comment:
question: if all the entities to copy are not related to other entities, would it be enough to just update the entity resourceId with the new one?

Reply:
@overbit, No, Because we always need to create a new field unrelated to the existing ones. 
This is for a case where the user wants to keep the source resource entities.